### PR TITLE
add code of conduct and contributing guide

### PR DESCRIPTION
**Pull request by pawelru**
closes https://github.com/insightsengineering/idr-tasks/issues/230 https://github.com/insightsengineering/idr-tasks/issues/229

This is intended to be general enough to be able to copy&paste into all other repos

Code of conduct is directly applied from https://www.contributor-covenant.org/ - no need to check the content. Please just think if that's the right choice
Contribution guidelines was written by me from scratch - therefore we might need some wordsmith here. It was written with the inspiration from https://mozillascience.github.io/working-open-workshop/contributing/

### CODE REVIEW

**Comment by dinakar29** [comment]
Awesome. Love it. I will propagate to all repos if and when this looks good to everyone else.

File: CODE_OF_CONDUCT.md
Comment (nikolas-burkoff) [inline CODE_OF_CONDUCT.md:21]: (optionally) 
```suggestion
* Demonstrating empathy and kindness towards other people
```

File: CONTRIBUTING.md
Comment (nikolas-burkoff) [inline CONTRIBUTING.md:31]:
```suggestion
For an introduction, please read [`README`](README.md) file. You can also find vignettes [here](vignettes/). Usually you will find a "Getting started" article. Please also see the package manual in the form of a `pkgdown` website.
```

File: CONTRIBUTING.md
Comment (nikolas-burkoff) [inline CONTRIBUTING.md:45]:
```suggestion
We kindly ask you to write a good issue with a minimal and reproducible example that demonstrates the problem. It would also be great to know your local environment (R session info) as well. You will find our template for issues when opening a bug ticket. This will speed up the fixing process a lot! 📈.
```

File: CONTRIBUTING.md
Comment (nikolas-burkoff) [inline CONTRIBUTING.md:49]:
```suggestion
If you notice a missing feature or you have an idea how to enhance existing functionalities - please let us know by creating a new issue. Before opening a new issue please double-check if it's already reported (but don't worry if occurs to be a duplicate - we will manage). If already there - up-vote ⬆️. We will have a look at it faster!
```

File: CONTRIBUTING.md
Comment (nikolas-burkoff) [inline CONTRIBUTING.md:51]:
```suggestion
Please elaborate on "why?" - what's the context, what's the benefit and for whom. You will find our template when opening a feature request ticket. This will help us prioritize and submit meaningful changes.
```

File: CONTRIBUTING.md
Comment (nikolas-burkoff) [inline CONTRIBUTING.md:61]: Are we opening the repo to write access or are we making external collaborators open a fork (I think the latter is more usual?) @dinakar29

File: CONTRIBUTING.md
Comment (nikolas-burkoff) [inline CONTRIBUTING.md:65]:
```suggestion
     Make sure your code passes all the checks and is well documented and tested so as not to decrease existing test coverage 💪.
```

File: CONTRIBUTING.md
Comment (nikolas-burkoff) [inline CONTRIBUTING.md:75]: Do you we want to give control to contributors to do the merge - feels like it's something we should do?

File: CONTRIBUTING.md
Comment (nikolas-burkoff) [inline CONTRIBUTING.md]: So this seems a little confusing as if you end up using staged.deps you explicitly do not follow the above branching convention

File: CONTRIBUTING.md
Comment (nikolas-burkoff) [inline CONTRIBUTING.md:92]: Link to action? Or say "see below"

File: CONTRIBUTING.md
Comment (nikolas-burkoff) [inline CONTRIBUTING.md:104]:
```suggestion
Although it allows for some flexibility - we recommend sticking to the style of the existing code. At the end of the day we want to have consistent codebase.
```

File: CONTRIBUTING.md
Comment (nikolas-burkoff) [inline CONTRIBUTING.md:106]:
```suggestion
Please note that there is a style and also `lintr` check in place that will validate your code.
```

**Review by nikolas-burkoff — COMMENTED** [review]
Some comments from me but looking good

File: CODE_OF_CONDUCT.md
Comment (pawelru) [reply CODE_OF_CONDUCT.md:21]: it's a copy&paste - not sure if we can introduce any changes

https://www.contributor-covenant.org/version/2/1/code_of_conduct/#:~:text=empathy%20and%20kindness-,toward,-other%20people

File: CONTRIBUTING.md
Comment (pawelru) [reply CONTRIBUTING.md]: once approved by any of the codeowners it should be ok

File: CONTRIBUTING.md
Comment (pawelru) [reply CONTRIBUTING.md:61]: I will rephrase it into the following
"If you don't have write access please fork it first."

File: CONTRIBUTING.md
Comment (pawelru) [inline CONTRIBUTING.md:123]: Does it make sense for you guys? Do you think we need it?
To me, there are some repo-specific rules e.g. in `tern` (specific naming convention, templating, etc.). I just want to open a door to specifying this while keeping the base identical across the repos. Will you use it or it's too much? @shajoezhu @gogonzo

File: CONTRIBUTING.md
Comment (pawelru) [inline CONTRIBUTING.md:122]: Does it make sense for you guys? Do you think we need it?
To me, there are some repo-specific rules e.g. in tern (specific naming convention, templating, etc.). I just want to open a door to specifying this while keeping the base identical across the repos. Will you use it or it's too much? @shajoezhu @gogonzo

File: CONTRIBUTING.md
Comment (shajoezhu) [inline CONTRIBUTING.md:3]:
```suggestion
First of all, thank you for taking the time to contribute! ✌️ 🎉
```

File: CONTRIBUTING.md
Comment (shajoezhu) [inline CONTRIBUTING.md:9]:
```suggestion
## Table of contents
```

File: CONTRIBUTING.md
Comment (shajoezhu) [inline CONTRIBUTING.md:114]:
```suggestion
The package maintainer also reserves the rights to manually adjust the criteria to recognize contributions.
```

File: CONTRIBUTING.md
Comment (shajoezhu) [reply CONTRIBUTING.md:122]: I wonder if we should also keep a copy of `CONTRIBUTING_ADDITIONAL.md` in this repo as well. Just to keep consistency. And suggest for removal if not needed.

Comment (gogonzo) [reply CONTRIBUTING.md:122]: Can't we just edit CONTRIBUTING.md in the future in the specific repository instead of making new files?

File: CONTRIBUTING.md
Comment (gogonzo) [inline CONTRIBUTING.md:101]: `spelling` is also used here

File: CONTRIBUTING.md
Comment (pawelru) [reply CONTRIBUTING.md:122]: > I wonder if we should also keep a copy of CONTRIBUTING_ADDITIONAL.md in this repo as well. Just to keep consistency. And suggest for removal if not needed.

I would rather have opt-in strategy than opt-out. There are too many repos to manage - we would forget about something.

> Can't we just edit CONTRIBUTING.md in the future in the specific repository instead of making new files?

Yeah - that's definitively an option. I was thinking about that and this would block future potential mass-updates of that file (if there would be a need). Therefore I suggested an additional, optional, repo-specific file. I know that this is something unusual but at least we would have a shared, master version that would be easy to update. Does that convince you or it's too much?

File: CONTRIBUTING.md
Comment (pawelru) [reply CONTRIBUTING.md:101]: ... and many others. But those are not focusing on styling the code (see header). I didn't want to list all of our checks as that list is too dynamic

File: CONTRIBUTING.md
Comment (pawelru) [reply CONTRIBUTING.md:122]: After I second thought I am going to remove that section because of YAGNI. Let's start with more simpler version and then we would add it when we need it.

**Review by dinakar29 — APPROVED** [review]

**Pull request merged**
